tests/integration-tests/tests/configure/test_pcluster_configure.py

```diff
@@ -161,3 +192,77 @@ def assert_config_contains_expected_values(
+    vpc_stack_public_subnet_id = vpc_stack.cfn_outputs["PublicSubnetId"]
+    # When there are omitted subnets, a note should be printed
+    omitted_note = "Note:  {0} subnet.+not listed.+".format(omitted_subnets_num) if omitted_subnets_num else ""
+    stage_list = [
+        {"prompt": r"AWS Region ID \[.*\]: ", "response": region},
+        {"prompt": r"EC2 Key Pair Name \[.*\]: ", "response": key_name},
+        {"prompt": r"Scheduler \[slurm\]: ", "response": scheduler},
+        {"prompt": r"Operating System \[alinux2\]: ", "response": os, "skip_for_batch": True},
+        {"prompt": fr"Minimum cluster size \({compute_units}\) \[0\]: ", "response": "1"},
+        {"prompt": fr"Maximum cluster size \({compute_units}\) \[10\]: ", "response": ""},
+        {"prompt": r"Master instance type \[t2\.micro\]: ", "response": instance},
+        {"prompt": r"Compute instance type \[t2\.micro\]: ", "response": instance, "skip_for_batch": True},
+        {"prompt": r"Automate VPC creation\? \(y/n\) \[n\]: ", "response": "n"},
+        {"prompt": r"VPC ID \[vpc-.+\]: ", "response": vpc_id},
+        {"prompt": r"Automate Subnet creation\? \(y/n\) \[y\]: ", "response": "n"},
+        {
+            "prompt": fr"{omitted_note}Master Subnet ID \[{vpc_stack_public_subnet_id}\]: ",
+            "response": headnode_subnet_id,
+        },
+        {
+            "prompt": fr"{omitted_note}Compute Subnet ID \[{vpc_stack_public_subnet_id}\]: ",
+            "response": compute_subnet_id,
+        },
+    ]
+    # When a user selects Batch as the scheduler, pcluster configure does not prompt for OS or compute instance type.
+    return [stage for stage in stage_list if scheduler != "awsbatch" or not stage.get("skip_for_batch")]
+
+
+@pytest.fixture()
+def subnet_in_use1_az3(vpc_stack):
+    # Subnet used to test the functionality of avoiding subnet in AZs that do not have user specified instance types
+    # Hard coding implementation to run on us-east-1 and create subnet in use1_az3 without c5.xlarge or m6g.xlarge
+    # Use use1_az3 as AZ, which does not have c5.xlarge or m6g.xlarge
+    # To-do: extend code to support arbitrary region and arbitrary instance types
+
+    # Verify that use1_az3 does not have c5.xlarge or m6g.xlarge.
+    ec2_client = boto3.client("ec2", region_name="us-east-1")
+    paginator = ec2_client.get_paginator("describe_instance_type_offerings")
+    page_iterator = paginator.paginate(
+        LocationType="availability-zone-id",
+        Filters=[
+            {"Name": "instance-type", "Values": ["c5.xlarge"]},
+            {"Name": "location", "Values": ["use1-az3"]},
+        ],
+    )
+    offerings = []
+    for page in page_iterator:
+        offerings.extend(page["InstanceTypeOfferings"])
+    logging.info(
+        "Asserting that c5.xlarge is not in use1-az3. If assertion fails, "
+        "c5.xlarge has been added to use1-az3, change the test code to test other rare instance type"
```

Comment:
Minor: I still think this comment is not clear. Especially don't like "change the test code to test other rare instance type".
If we want to provide information, we should make clear the exact case we are testing here. But I don't think we should suggest something like `test other rare instance type` because we could also change AZ for example. In any case, we should let the future maintainers decide path forward and not mislead them into thinking we need to run this test on some expensive or rare instance type